frontend/src/store/pinia-modules/sqlEditor.ts

```diff
@@ -0,0 +1,233 @@
+  }),
+
+  getters: {
+    connectionTreeByInstanceId(state): Partial<ConnectionAtom> {
+      const idx = state.connectionTree.findIndex((item) => {
+        return item.id === state.connectionContext.instanceId;
+      });
+
+      return idx !== -1 ? state.connectionTree[idx] : {};
+    },
+
+    connectionInfo() {
+      const projectStore = useProjectStore();
+      const instanceStore = useInstanceStore();
+      const databaseStore = useDatabaseStore();
+      const tableStore = useTableStore();
+
+      return {
+        projectListById: projectStore.projectById,
+        instanceListById: instanceStore.instanceById,
+        databaseListByInstanceId: databaseStore.databaseListByInstanceId,
+        databaseListByProjectId: databaseStore.databaseListByProjectId,
+        tableListByDatabaseId: tableStore.tableListByDatabaseId,
+      };
+    },
+    connectionInfoByInstanceId() {
+      let instance = {} as any;
+      let databaseList: Database[] = [];
+      let tableList = [];
+
+      if (!isEmpty(this.connectionTreeByInstanceId)) {
+        instance = this.connectionTreeByInstanceId;
+        databaseList = useDatabaseStore().getDatabaseListByInstanceId(
+          instance.id
+        );
+
+        tableList = instance.children
+          .map((item: ConnectionAtom) =>
+            useTableStore().getTableListByDatabaseId(item.id)
+          )
+          .flat();
+      }
+
+      return {
+        instance,
+        databaseList,
+        tableList,
+      };
+    },
+    findProjectIdByDatabaseId:
+      (state) =>
+      (databaseId: DatabaseId): ProjectId => {
+        const databaseStore = useDatabaseStore();
+        let projectId = UNKNOWN_ID;
+        const databaseListByProjectId = databaseStore.databaseListByProjectId;
+        for (const [id, databaseList] of databaseListByProjectId) {
+          const idx = databaseList.findIndex(
+            (database: Database) => database.id === databaseId
+          );
+          if (idx !== -1) {
+            projectId = id;
+            break;
+          }
+        }
+        return projectId;
+      },
+
+    currentSlug(state) {
+      const connectionContext = state.connectionContext;
+      return `${connectionContext.instanceId}/${connectionContext.databaseId}/${connectionContext.tableId}`;
+    },
+    /**
+     * check the connection whether disconnected
+     * 1、If the context is not set the instanceId, return true
+     * 2、If the context is set the instanceId, but not set the databaseId and databaseType is not MYSQL or TIDB, return true
+     * @param state
+     * @returns boolean
+     */
+    isDisconnected(state) {
+      const ctx = state.connectionContext;
+      return (
+        ctx.instanceId === UNKNOWN_ID ||
+        (ctx.databaseId === UNKNOWN_ID &&
+          ctx.databaseType !== "MYSQL" &&
+          ctx.databaseType !== "TIDB")
+      );
+    },
+  },
+
+  actions: {
+    setSqlEditorState(payload: Partial<SqlEditorState>) {
```

Comment:
`setSQLEditorState`

Reply:
Maybe rename the variable in a split PR, in here, only migrate the old one to pinia.